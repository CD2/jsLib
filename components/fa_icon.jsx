import React from 'react'
import PropTypes from 'prop-types'
import {styled} from "utils/theme/index"
import decorate from 'utils/decorate'

export default class FaIcon extends React.Component {

  static PropTypes = {
    icon: PropTypes.string.isRequired,
    hoverColor: PropTypes.string,
    color: PropTypes.string,
    size: PropTypes.number,
    onClick: PropTypes.func,
  }

  get className() {
    let { className, icon } = this.props
    className += ` fa fa-${icon}`
    return className
  }

  render() {
    const { onClick } = this.props
    return (
      <i className={this.className} onClick={onClick}/>
    )
  }
}
export default decorate(
  styled`
    ${({ color, theme, size }) => {
      const col = color || theme.darkBackground
      const sze = size || 1
      return (
        `
        color: ${col};
        font-size: ${sze}em;
        `
      )
    }}
    ${({ hoverColor }) => {
      if(hoverColor) {
        return (
          `
          &:hover{
            color: ${hoverColor};
          }
          `
        )
      }
    }}
  `,
  FaIcon
)
