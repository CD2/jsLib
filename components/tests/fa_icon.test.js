import t from 'utils/test_helper'

import { FaIcon } from '../fa_icon'

t.describe_component(FaIcon, { icon:`test_icon`, className:`test` }, (c) => {
  describe(`proptypes`, () => {
    c.accepts_prop(`icon`)
    c.accepts_prop(`hoverColor`)
    c.accepts_prop(`color`)
    c.accepts_prop(`size`)
  })

  c.has_css(`i`)
})
