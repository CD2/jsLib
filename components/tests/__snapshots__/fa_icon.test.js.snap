// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <FaIcon /> renders correctly 1`] = `"<i class=\\"fa fa-test_icon test\\"></i>"`;
